src/backend/cdb/cdbllize.c

```diff
@@ -186,7 +186,7 @@ cdbparallelize(PlannerInfo *root,
 	switch (query->commandType)
 	{
 		case CMD_SELECT:
-			if (query->intoClause != NULL)
+			if (query->isCtas)
 			{
```

Comment:
will this impact other queries which will use intoClause?

Reply:
The intoClause went away in the 9.2 merge, but was put back in order to allow cdbparallelize to understand that it needs to send the tuples into a relation.

I believe that it also uses this code path to identify `SELECT INTO` commands, as they behave identically in upstream. In that case the `isCtas` variable name may be slightly misleading.

Reply:
'SELECT INTO' is transformed to CreateTableAsStmt earlier in transformTopLevelStmt().

isCtas seems to be a bit misleading. isForCtas, or just force_parallel to be more common?

Reply:
@yydzero This would not impact other queries. `intoClause` has been removed by PG 9.2, we add it back here to correctly handle dispatch issue. Currently only CTAS (CREATE TABLE AS / SELECT INTO) statement would use `intoClause`, for other queries, it would be set to NULL.

Reply:
@tvesely Yes. We need a more proper name here.